api/app/Actions/Profile/UpdateProfileAction.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Actions\Profile;
+
+use App\Exceptions\Profile\ProfileNotFoundException;
+use App\Repositories\User\UserRepository;
+use Illuminate\Auth\AuthenticationException;
+use Illuminate\Support\Facades\Auth;
+
+final class UpdateProfileAction
+{
+    private UserRepository $profileRepository;
+
+    public function __construct(UserRepository $profileRepository)
+    {
+        $this->profileRepository = $profileRepository;
+    }
+
+    public function execute(UpdateProfileRequest $request): UpdateProfileResponse
+    {
+        $user = Auth::user();
+
+        if (!$user) {
+            throw new AuthenticationException();
+        }
+
+        $profile = $this->profileRepository->getById($user->id);
+
+        if (!$profile) {
+            throw new ProfileNotFoundException();
+        }
+
+        $profile->avatar = $request->getAvatar();
+        $profile->name = $request->getName();
+        $profile->welcome_message = $request->getWelcomeMessage();
+        $profile->language = $request->getLanguage() ?: 'en';
+        $profile->date_format = $request->getDateFormat() ?: 'american';
```

Comment:
The default values must not be set here. There are two ways of updating record: passing all the fields and setting them (`POST` HTTP method) or passing some fields and update only them, remain fields stay unchanged (`PUT` HTTP method). We are going to use the second one, as we have different tabs.